app/src/main/java/org/kiwix/kiwixmobile/language/LanguageFragment.kt

```diff
@@ -105,26 +120,27 @@ class LanguageActivity : BaseActivity() {
     Saving -> Unit
   }
 
-  override fun onCreateOptionsMenu(menu: Menu): Boolean {
-    menuInflater.inflate(R.menu.menu_language, menu)
+  override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
+    super.onCreateOptionsMenu(menu, inflater)
+    inflater.inflate(R.menu.menu_language, menu)
     val search = menu.findItem(R.id.menu_language_search)
     (search.actionView as SearchView).setOnQueryTextListener(SimpleTextListener {
       languageViewModel.actions.offer(Filter(it))
     })
-    return true
   }
 
   override fun onOptionsItemSelected(item: MenuItem): Boolean {
-    when (item.itemId) {
+    val returnValue = when (item.itemId) {
       android.R.id.home -> {
-        onBackPressed()
+        requireActivity().onBackPressed()
         return true
```

Comment:
this return should probably be omitted

Reply:
but also this could be written to 
```
return when(id){
 R.id.thing ->{ //some stuff 
true}
else -> super.OnOptionsItemSelected
}
```

right?

Reply:
yes, much better!